feat: add project-level permissions docs (#27887)
supabase/supabase: apps/docs/content/guides/platform/access-control.mdx

---
title: 'Access Control'
description: 'Roles and permissions at the organization and project levels'
---

Supabase provides granular access controls to manage permissions across your organizations and projects.

For each organization and project, a member can have one of the following roles:

- **Owner**: full access to everything in organization and project resources.

> - **Administrator**: full access to everything in organization and project resources except updating organization settings, transferring projects outside of the organization, and adding new owners.

Reviewer: Admins can't add owners - but they can add new members

```suggestion
- **Administrator**: full access to everything in organization and project resources except updating organization settings, transferring projects outside of the organization, and adding new members.
```

Reviewer 2: @joshenlim that's what is conveyed here - _except_ updating org settings, ... adding new owners. Feel free to change the structure if thats not clear.

Reviewer: ohhhhh okay gotcha on this!

- **Developer**: read-only access to organization resources and content access to project resources but cannot change any project settings.
- **Read-Only**: read-only access to organization and project resources.

<Admonition type="note">

> Owner, Administrator, and Developer roles are available on all projects but roles, such as Read-Only, are only available on the [Team and Enterprise plans](https://supabase.com/pricing).

Reviewer 3: This was so hard to read first time I read it. 
Can it not just say 

```suggestion
Read-only and Billing roles only available on [team and enterprise plans](https://supabase.com/pricing).

```

Author: good suggestion, will do

Author: we don't want to mention Billing role yet

</Admonition>

When you first create an account, a default organization is created for you and you'll be assigned as the **Owner**. Any organizations you create will assign you as **Owner** as well.

## Manage organization members

To invite others to collaborate, visit your organization's team [settings](/dashboard/org/_/team) to send an invite link to another user's email. The invite is valid for 24 hours.

<Admonition type="note">

Invites sent from a SAML SSO account can only be accepted by another SAML SSO account from the same identity provider.

This is a security measure to prevent accidental invites to accounts not managed by your enterprise's identity provider.

</Admonition>

### Transferring ownership of an organization

Each Supabase organization must have at least one owner. If your organization has other owners then you can relinquish ownership and leave the organization by clicking **Leave team** in your organization's team [settings](/dashboard/org/_/team).

Otherwise, you'll need to invite a user as **Owner**, and they need to accept the invitation, or promote an existing organization member to **Owner** before you can leave the organization.

> ### Organization Roles vs Project Roles

Reviewer 2: ```suggestion
### Organization Scoped Roles vs Project Scoped Roles
```

> Each member in the organization can be assigned a role at either the organisation level, or for one or more specific projects. If the member has a role on the organization, the member will have the equivalent permissions for that role across all projects under the organization. However, if you might want to only allow access to specific projects for a member, you may assign roles for the member on the projects that you want to permit access to.

Reviewer 2: ```suggestion
Each member in the organization can be assigned a role scoped to the organization or to specific projects. If the member has a role at the organization level, they will have the equivalent permissions for that role across all current and future projects in the organization. 

With project scoped permissions, you can assign members to roles scoped to specific projects. 
```

<Admonition type="note">

> Project level roles are only available on the [Enterprise plan](https://supabase.com/pricing)

Reviewer 2: This can be moved to just below the heading

</Admonition>

### Organization permissions across roles

> The table below shows the corresponding organization permissions, by resource and action, for each available role you can assign an organization member in Studio.

Reviewer 2: ```suggestion
The table below shows the actions each role can take on the resources belonging to the organization.
```

| **Audit Logs**                  |            |                                         |                                         |                                         |                                         |
| View Audit logs                 | -          | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> |
| **Legal Documents**             |            |                                         |                                         |                                         |                                         |
| SOC2 Type 2 Report              | Download   | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> |
| Security Questionnaire          | Download   | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> |

### Project permissions across roles

> The table below shows the corresponding project permissions, by resource and action, for each available role you can assign an organization member in Studio.

Reviewer 2: ```suggestion
The table below shows the actions each role can take on the resources belonging to the project.
```

| Disable branching                | -                      | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> |           <IconX size={14} />           |                       <IconX size={14} />                       |
|                                  | Create                 | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> |                       <IconX size={14} />                       |
|                                  | Delete                 | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> |                       <IconX size={14} />                       |
|                                  | List                   | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> |             <IconCheck size={14} color="#3FCF8E" />             |

[^1]: Available on the Team and Enterprise Plans.
[^2]:
    Invites sent from a SSO account can only be accepted by another SSO account

>     coming from the same identity provider. This is a security measure that

Reviewer 2: there seems to be a line break here on the preview

    prevents accidental invites to accounts not managed by your company's
    enterprise systems.

[^3]: Available on the Team and Enterprise Plans.

> [^4]: Only available on projects using PostgreSQL 14 and above. You can upgrade your project through [infrastructure settings](https://supabase.com/dashboard/project/_/settings/infrastructure).

Reviewer 2: This limitation can be mentioned on just the read replica page. needn't be on the permissions page.

[^5]: Read-Only role is able to access secrets.

> [^6]: Only select queries.

Reviewer 2: Limited to executing SELECT queries

|                                 | Update   | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> |           <IconX size={14} />           |           <IconX size={14} />           |
| Payment Methods                 | View     | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> |
|                                 | Update   | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> |           <IconX size={14} />           |           <IconX size={14} />           |
| Usage                           | View     | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> |
| **Integrations (Org Settings)** |          |                                         |                                         |                                         |                                         |

> | Authorize GitHub                | -        | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> |           <IconX size={14} />           | <IconCheck size={14} color="#3FCF8E" /> |

Reviewer 4: Is this correct? Seems to run against my understanding of "read-only"

<img width="866" alt="image" src="https://github.com/supabase/supabase/assets/26616127/2289375c-2ba2-4493-a45f-c07fea0552c5">

Author: you're right! will update.

| **Audit Logs**                  |          |                                         |                                         |                                         |                                         |
| View Audit logs                 | -        | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> |
| **Legal Documents**             |          |                                         |                                         |                                         |                                         |
| SOC2 Type 2 Report              | Download | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> |
| Security Questionnaire          | Download | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> | <IconCheck size={14} color="#3FCF8E" /> |

> ### Project permissions across roles

Reviewer 4: Might need a short explanation of how this table differs from the previous one, got a bit confused

Author: good call, will add some color
